staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -745,6 +745,11 @@ func (c *Cacher) GuaranteedUpdate(
 
 // Count implements storage.Interface.
 func (c *Cacher) Count(pathPrefix string) (int64, error) {
+	keys := c.watchCache.ListKeys()
```

Comment:
First - that should not read from cache until watchcache isn't initialized.

But even with this change, I'm slightly worried about staleness of this data.
Technically this is pretty much used to expose the metric, which is gathered every 1m, so some delay is probably acceptable. But as long as we don't even know how much into the past we're, this seems too risky.

We're generally using metrics as a way to check if the system works correctly, so this seems like a risky change to me.

Reply:
>First -  that should not read from cache until watchcache isn't initialized.

Yes, it should check watchcache's status.

>But even with this change, I'm slightly worried about staleness of this data.
Technically this is pretty much used to expose the metric, which is gathered every 1m, so some delay is probably acceptable. But as long as we don't even know how much into the past we're, this seems too risky.

In large-scale scenarios, the count requests would block other raft requests when locked treeIndex in ETCD. 
It's usefully when resource of kubernetes is over 2 million objects.